src/engraving/rendering/score/paint.cpp

```diff
@@ -122,6 +122,12 @@ void Paint::paintScore(Painter* painter, Score* score, const IScoreRenderer::Pai
                 painter->translate(-pageRect.topLeft());
             }
 
+            if (painter->hasClipping() && (drawRect.top() < pageRect.top() || drawRect.bottom() > pageRect.bottom()
```

Comment:
Shouldn't it be drawRect.top() **>** pageRect.top() || drawRect.bottom() **<** pageRect.bottom()?

Reply:
Or maybe we can even use something like drawRect.intersected(pageRect) here...

Reply:
If you merged that other change you mentioned to draw the cursor separately then yes, I'd like to try the intersect method again.  But no, this is otherwise correct - if the drawRect extends outside the pageRect, it uses the whole pageRect for clipping.

Reply:
Yes, I merged it

Reply:
Something's changed since I last worked on this, I can't get the clipping rectangle to work properly at all now...happy to hand it over as I don't know how much time I'll have.  The problem definitely still exists at any rate.

Reply:
Actually...I did get it working...but the playback cursor still draws off the edge of the page area which looks a bit odd...